### PR TITLE
fix remaining issues in #94

### PR DESCRIPTION
**Pull request by rhaschke**
We missed two issues before merging #94:

* assignment doesn't work with `==`
* Switching to the correct `WallDuration` interface for `MoveGroupInterface`, made the constructors ambigue if not the full signature is specified. Hence, I propose to remove the default values for the old constructor. This will automatically choose the new version, which is preferred anyway.

Please also cherry-pick into Jade. #94 was not yet merged into Indigo. I will integrate this patch into #144 as well, which is the corresponding PR for Indigo.

### CODE REVIEW

File: moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group.h [hunk 1]
```diff
@@ -121,8 +121,8 @@ class MoveGroup
     */
   MoveGroup(const Options &opt, const boost::shared_ptr<tf::Transformer> &tf = boost::shared_ptr<tf::Transformer>(),
             const ros::WallDuration &wait_for_servers = ros::WallDuration());
-  MOVEIT_DEPRECATED MoveGroup(const Options &opt, const boost::shared_ptr<tf::Transformer> &tf = boost::shared_ptr<tf::Transformer>(),
-                           const ros::Duration &wait_for_servers = ros::Duration());
+  MOVEIT_DEPRECATED MoveGroup(const Options &opt, const boost::shared_ptr<tf::Transformer> &tf,
+                              const ros::Duration &wait_for_servers);
```
Comment (davetcoleman) [inline moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group.h]: does this break API?

**Comment by v4hn** [comment]
This is neither an API nor an ABI problem, nifty.

LGTM +1

File: moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group.h (hunk 1 above)
Comment (rhaschke) [inline moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group.h]: According to [abi compliance checker](http://ispras.linuxbase.org/index.php/ABI_compliance_checker) this is fine: [REPORT](https://rawgit.com/rhaschke/03afceeee1e30779cbf0bcd1b27f0419/raw/87f7a0cae43c5b505a61cf0f15c46864a3290985/compat_report.html)

**Comment by rhaschke** [comment]
Cherry-picked into Jade.

**Pull request merged**